operator/exporter.py

```diff
@@ -0,0 +1,266 @@
+
+    try:
+        resp = execute(cmd)
+        config_data = json.loads(resp.stdout)
+
+        data = config_data['data']
+
+        key_data = []
+        brick_data = {}
+
+        for key, value in data.items():
+            if key.endswith('info'):
+                key_data.append(key)
+
+        for key in key_data:
+            storage_data = json.loads(data[key])
+            brick_data[key.rstrip(".info")] = storage_data.get("bricks")
+            #brick_data.append(storage_data.get("bricks"))
+
+        return brick_data
+
+    except CommandError as err:
+        logging.error(logf(
+            "Failed to get brick data from configmap",
+            error=err
+        ))
+        return None
+
+
+def collect_all_metrics():
+    """ Collect all metrics data from different listening pods """
+
+    metrics = Metrics()
+
+    metrics = get_operator_data(metrics)
+
+    pod_data = get_pod_data()
+
+    for pod_name, pod_details in pod_data.items():
+
+        #Skip GET request to operator
+        if "operator" in pod_name:
+            metrics.operator.update({"pod_name": pod_name})
+            metrics.operator.update(pod_details)
+            continue
+
+        response = http.request('GET', 'http://'+ pod_details["ip_address"] +':8050/_api/metrics')
+        if response.status == 200:
+
+            if "nodeplugin" in pod_name:
+
+                metrics.nodeplugins.append(json.loads(response.data)["pod"])
+                # Is for reqd here?? Maybe yes... Multiple node not tested
+                for nodeplugin in metrics.nodeplugins:
+                    if nodeplugin["pod_name"] in pod_name:
+                        nodeplugin.update(pod_details)
+
+            if "provisioner" in pod_name:
+
+                #storage_pool_name = json.loads(r.data)["storages"]["name"]
+                #metrics.storages[storage_pool_name] = json.loads(r.data)["storages"]
+                metrics.storages = json.loads(response.data)["storages"]
+                metrics.provisioner.update({"pod_name": pod_name})
+                metrics.provisioner.update(json.loads(response.data)["pod"])
+                metrics.provisioner.update(pod_details)
+
+            if "server" in pod_name:
+
+                brick_data = collect_brick_data()
+                for storage in metrics.storages:
+
+                    if storage["name"] in pod_name:
+                        if not storage.get("bricks"):
+                            storage.update({"bricks": brick_data[storage["name"]]})
+
+                        for brick in storage["bricks"]:
+                            brick_name = brick["node"].rstrip("."+storage["name"])
+
+                            if brick_name in pod_name:
+                                # Is pod name required here?
+                                brick.update(json.loads(response.data)["pod"])
+                                brick.update(pod_details)
+        else:
```

Comment:
@aravindavk Any comments on marking the pods as down when,
1) When pod IP has not been assigned due non-deployment of service
2) When response status is not 200

Reply:
You can mark Offline for both the cases